components/shared/ui/appPage.tsx

```diff
@@ -5,31 +5,79 @@ import { perCollectionSEOTitle } from "../../../lib/constants/labels";
 import { ValidCollectionCodename } from "../../../lib/types/perCollection";
 import { useSmartLink } from "../../../lib/useSmartLink";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
-import { Block_Navigation } from "../../../models";
+import { Article, Block_Navigation, Product, SEOMetadata, WSL_WebSpotlightRoot } from "../../../models";
 import { SiteCodenameProvider } from "../siteCodenameContext";
 import { Footer } from "./footer";
 import { Menu } from "./menu";
 
 type Props = Readonly<{
   children: ReactNode;
   siteCodename: ValidCollectionCodename;
-  itemId?: string;
+  item?: WSL_WebSpotlightRoot | Article | Product;
```

Comment:
You might also want to consider making this required too (you would have to widen the type to also include `Page`). It will be easier to add smartlink buttons for page elements when all pages will have the itemId.

Reply:
set to required, adjusted types